query-planner-js/src/composedSchema/__tests__/removeInaccessibleElements.test.ts

```diff
@@ -0,0 +1,120 @@
+import { buildSchema } from 'graphql';
+import { removeInaccessibleElements } from '../removeInaccessibleElements';
+
+describe('removeInaccessibleElements', () => {
+  it(`removes @inaccessible fields`, () => {
+    let schema = buildSchema(`
+      directive @core(feature: String!) repeatable on SCHEMA
+
+      directive @inaccessible on FIELD_DEFINITION | OBJECT | INTERFACE | UNION
+
+      schema
+        @core(feature: "https://specs.apollo.dev/core/v0.1")
+        @core(feature: "https://specs.apollo.dev/inaccessible/v0.1")
+      {
+        query: Query
+      }
+
+      type Query {
+        someField: String
+        privateField: String @inaccessible
+      }
+    `);
+
+    schema = removeInaccessibleElements(schema);
+
+    const queryType = schema.getQueryType()!;
+
+    expect(queryType.getFields()['someField']).toBeDefined();
+    expect(queryType.getFields()['privateField']).toBeUndefined();
+  });
+
+  it(`removes @inaccessible object types`, () => {
+    let schema = buildSchema(`
+      directive @core(feature: String!) repeatable on SCHEMA
+
+      directive @inaccessible on FIELD_DEFINITION | OBJECT | INTERFACE | UNION
+
+      schema
+        @core(feature: "https://specs.apollo.dev/core/v0.1")
+        @core(feature: "https://specs.apollo.dev/inaccessible/v0.1")
+      {
+        query: Query
+      }
+
+      type Query {
+        fooField: Foo
+      }
+
+      type Foo @inaccessible {
```

Comment:
Note that if you add a union type here that has member `Foo`, object type `Foo` won't be removed, since there will be a reference to `Foo` in the members clause of the union type. (This is similar to the issue with interfaces and `implements` clauses.)

It would be good to modify this test to add such a union type, and to change `toAPISchema()` to remove `@inaccessible` object types from union member clauses, since there's no way to mark a member clause with `@inaccessible` like there is for fields. (This again goes back to [a comment you made in this Quip thread](https://apollographql.quip.com/pr0TAq6MDgmC/Contracts-Inaccessible-propagation#KYUACAdxZRj)).

Reply:
Looks like this is resolved via inclusion of `union Bar = Foo` now?